contracts/ethereum/v0.7/src/KeeperConsumerBenchmark.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity 0.7.6;
+
+contract KeeperConsumerBenchmark {
+  event PerformingUpkeep(address from, uint256 initialCall, uint256 nextEligible, uint256 blockNumber);
+
+  uint256 public initialCall = 0;
+  uint256 public nextEligible = 0;
+  uint256 public testRange;
+  uint256 public averageEligibilityCadence;
+  uint256 public checkGasToBurn;
+  uint256 public performGasToBurn;
+  mapping(bytes32 => bool) public dummyMap; // used to force storage lookup
+
+  uint256 public count = 0;
+
+  constructor(uint256 _testRange, uint256 _averageEligibilityCadence, uint256 _checkGasToBurn, uint256 _performGasToBurn) {
+    testRange = _testRange;
+    averageEligibilityCadence = _averageEligibilityCadence;
+    checkGasToBurn = _checkGasToBurn;
+    performGasToBurn = _performGasToBurn;
+  }
+
+  function checkUpkeep(bytes calldata data) external view returns (bool, bytes memory) {
+    uint256 startGas = gasleft();
+    bytes32 dummyIndex = blockhash(block.number - 1);
+    bool dummy;
+    // burn gas
+    while (startGas - gasleft() < checkGasToBurn) {
```

Comment:
I'm thinking whether we should add some buffer here to ensure checkUpkeep doesn't take more than checkGasToBurn? (e.g. https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.7/tests/UpkeepMock.sol#L12-L32) 

Not a strong opinion but could be helpful in debugging

Reply:
Since this was working fine for tests done until now, I think it is best to not make this change. Moreover, any changes would mean the test results would be invalidated.